File: 0_1.baekjoon/brute-force/java/src/BOJ1747.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class BOJ1747 {
    private static int N;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());

        int answer = - 1;

        while (answer == -1) {
            if (isPrime(N) && isPalindrome(N)) {
                answer = N;
            }
            N++;
        }
        System.out.println(answer);
    }

    private static boolean isPalindrome(int num) {
        String strNum = String.valueOf(num);

        int left = 0, right = strNum.length() - 1;

        while (left < right) {
            if (strNum.charAt(left) != strNum.charAt(right)) {
                return false;
            }

            left++;
            right--;
        }
        return true;
    }

    private static boolean isPrime(int num) {
        if (num <= 1) {
            return false;
        }

        int sqrt = (int) Math.sqrt(num);

        for (int i = 2; i <= sqrt; i++) {
            if (num % i == 0) {
                return false;
            }
        }

        return true;
    }
}

File: 0_1.baekjoon/brute-force/kotlin/src/BOJ1747.kt
import kotlin.math.sqrt

fun main() = with(System.`in`.bufferedReader()) {
    var n = readLine()!!.toInt()

    var answer = -1

    while (answer == -1) {
        if (isPrime(n) && isPalindrome(n)) {
            answer = n
        }
        n++
    }

    println(answer)
}

private fun isPrime(num: Int): Boolean {
    if (num <= 1) {
        return false
    }

    val sqrt = sqrt(num.toDouble()).toInt()

    for (i in 2..sqrt) {
        if (num % i == 0)
            return false
    }

    return true
}

private fun isPalindrome(num: Int): Boolean {
    val strNum = num.toString()

    var left = 0
    var right = strNum.length - 1

    while (left < right) {
        if (strNum[left++] != strNum[right--]) {
            return false
        }
    }

    return true
}